src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -332,17 +346,26 @@ private function getModulesTree($lang, $type, $selected, $search = null)
      * @param $catalogue
      * @param $type
      * @param $selected
-     * @param null $search
+     * @param string|null $search
+     * @param string|null $module
      *
      * @return array
      */
-    private function getCleanTree(TreeBuilder $treeBuilder, $catalogue, $type, $selected, $search = null)
+    private function getCleanTree(TreeBuilder $treeBuilder, $catalogue, $type, $selected, $search = null, $module = null)
     {
-        $selected = ('mails' === $type && 'subject' === $selected ? false : $selected);
-
         $translationsTree = $treeBuilder->makeTranslationsTree($catalogue);
-        $translationsTree = $treeBuilder->cleanTreeToApi($translationsTree, $this->container->get('router'), $selected, $search);
+        $translationsTree = $treeBuilder->cleanTreeToApi($translationsTree, $this->container->get('router'), $this->getSelectedTheme(), $search, $module);
 
         return $translationsTree;
     }
+
+    /**
+     * @return \PrestaShop\PrestaShop\Core\Addon\Theme\Theme
+     *
+     * @todo: When the theme will be selectable in the form, this function should be updated accordingly.
```

Comment:
ping @kpodemski, once the first form will be updated with a new field named "theme", only this function needs to be updated and the translations manager "should" be able to translate according to the specified theme 👍